core/modules/global/ResourceUnit.cc

```diff
@@ -159,26 +169,33 @@ ResourceUnit::_setFromPath(std::string const& path) {
             _unitType = GARBAGE;
             return;
         }
+        if(_markGarbageIfDone(t)) { return; }
         t.next();
+        if(t.token().empty()) { _unitType = GARBAGE; return; }
         _chunk = t.tokenAsInt();
         _ingestLeafAndKeys(t.token());
     } else if(rTypeString == prefix(CQUERY)) {
         // Import as chunk query
         _unitType = CQUERY;
+        if(_markGarbageIfDone(t)) { return; }
         t.next();
         _db = t.token();
         if(_db.empty()) {
             _unitType = GARBAGE;
             return;
         }
+        if(_markGarbageIfDone(t)) { return; }
         t.next();
+        if(t.token().empty()) { _unitType = GARBAGE; return; }
         _chunk = t.tokenAsInt();
         _ingestLeafAndKeys(t.token());
 
     } else if(rTypeString == prefix(RESULT)) {
         _unitType = RESULT;
+        if(_markGarbageIfDone(t)) { return; }
         t.next();
         _hashName = t.token();
+        if(_hashName.empty()) { _unitType = GARBAGE; return; }
```

Comment:
(Or, maybe this kind of triviality is not the best thing to be spending your time on right now, ha!  This has tests, is not so bad, and I'm fine if you want to roll this in.)